In [1]:
import pandas as pd

In [2]:
pd.set_option('display.max_columns',None)

In [4]:
players = pd.read_csv("../raw_data/players.csv",index_col=0)
players.head()

height  weight   birthDate             collegeName position  \
nflId                                                                 
2539334     72     190  1990-09-10              Washington       CB   
2539653     70     186  1988-11-01  Southeastern Louisiana       CB   
2543850     69     186  1991-12-18                  Purdue       SS   
2555162     73     227  1994-11-04         Louisiana State      MLB   
2555255     75     232  1993-07-01               Minnesota      OLB   

                displayName  
nflId                        
2539334     Desmond Trufant  
2539653       Robert Alford  
2543850       Ricardo Allen  
2555162         Deion Jones  
2555255  De'Vondre Campbell

In [5]:
players["height"].unique()

array(['72', '70', '69', '73', '75', '6-0', '78', '77', '68', '66', '71',
       '76', '6-5', '74', '5-11', '5-9', '6-1', '5-10', '6-4', '79',
       '6-3', '6-2', '67', '5-6', '5-8', '80', '6-6', '6-7', '5-7'],
      dtype=object)

Como podemos ver, para la altura hay ciertos valores con un guión en medio, que indicara los pies y las pulgadas

In [6]:
number_players = 0
for index,row in players.iterrows():
    if "-" in row["height"]:
        number_players += 1

In [7]:
number_players

240

In [8]:
len(players)

1303

De 1303 jugadores que tiene el dataset, 240 tiene un guión en medio. Hay que saber cual es la medida y convertirla de manera adecuada.

In [11]:
players[players["height"]=="72"]

height  weight   birthDate          collegeName position  \
nflId                                                              
2539334     72     190  1990-09-10           Washington       CB   
2552600     72     198  1993-05-24  Southern California       WR   
79848       72     204  1987-12-20           Ohio State       SS   
2555383     72     191  1994-04-06      Louisiana State       CB   
2557958     72     181  1996-05-21           Washington       CB   
...        ...     ...         ...                  ...      ...   
2552317     72     232  1991-05-25           Penn State      MLB   
2561403     72     205  1995-08-08         Fresno State       WR   
2560875     72     240  1997-04-20       North Carolina       LB   
2556273     72     230  1993-12-01     Eastern Michigan       RB   
2561359     72     180  1995-03-19       Texas Southern       WR   

             displayName  
nflId                     
2539334  Desmond Trufant  
2552600   Nelson Agholor  
79848    Malcolm Jenkins  
2555383      Jalen Mills  
2557958     Sidney Jones  
...                  ...  
2552317        Mike Hull  
2561403    Da'Mari Scott  
2560875      Andre Smith  
2556273   Darius Jackson  
2561359     Darvin Kidsy  

[153 rows x 6 columns]

Pongamos el ejemplo de Nelson Agholor. Mide 1.83 (6 pies). Según el dataset, mide 72 pulgadas, esto es correcto ya que 1 pie = 12 pulgadas. Por tanto, si no hay guión se está indicando la altura en pies.

In [12]:
players[players["height"]=="6-5"]

height  weight   birthDate           collegeName position  \
nflId                                                               
2507828    6-5     268  02/26/1987                  UCLA       TE   
2543471    6-5     245  02/05/1991         Florida State       WR   
2552580    6-5     240  09/20/1991              Arkansas       TE   
2543683    6-5     262  09/29/1992            Washington       TE   
2505996    6-5     218  01/03/1981           Mississippi       QB   
2532835    6-5     255  10/03/1988                   USC       TE   
2553658    6-5     255  03/29/1991                Purdue       TE   
2495893    6-5     232  03/23/1984       Central Florida       WR   
497238     6-5     260  06/16/1988              Oklahoma       TE   
2556543    6-5     265  02/04/1994  North Carolina State       TE   
71435      6-5     251  06/11/1987              Virginia       TE   

                     displayName  
nflId                             
2507828            Logan Paulsen  
2543471          Kelvin Benjamin  
2552580               A.J. Derby  
2543683  Austin Seferian-Jenkins  
2505996              Eli Manning  
2532835            Rhett Ellison  
2553658              Gabe Holmes  
2495893         Brandon Marshall  
497238          Jermaine Gresham  
2556543           David Grinnage  
71435              John Phillips

Eli Manning, por ejemplo mide 1.96, 6 pies y 5 pulgadas. 1 pie son 12 pulgadas y una pulgada son 0,0254

In [13]:
players_copy = players.copy()

In [27]:
def changeHeight(oldHeight):
    if "-" in oldHeight:
        height = oldHeight.split("-")
        feet = height[0]
        inches = height[1]
        
        value = (int(feet) * 12 + int(inches)) * 0.0254
    else:
        value = int(oldHeight) * 0.0254
        
    value = round(value,2)
    
    return value

In [28]:
changeHeight("6-5")

1.96

In [29]:
players["height"] = players["height"].apply(changeHeight)

In [30]:
players

height  weight   birthDate             collegeName position  \
nflId                                                                  
2539334    1.83     190  1990-09-10              Washington       CB   
2539653    1.78     186  1988-11-01  Southeastern Louisiana       CB   
2543850    1.75     186  1991-12-18                  Purdue       SS   
2555162    1.85     227  1994-11-04         Louisiana State      MLB   
2555255    1.91     232  1993-07-01               Minnesota      OLB   
...         ...     ...         ...                     ...      ...   
2553798    1.88     235  09/17/1991              Vanderbilt       LS   
2556814    1.88     200  05/17/1994                    Duke       WR   
2552576    1.98     231  1992-04-25            Oregon State       QB   
2561469    1.75     187  1993-03-15      Alabama-Birmingham       CB   
2558095    1.88     196  10/12/1995                     LSU       WR   

                displayName  
nflId                        
2539334     Desmond Trufant  
2539653       Robert Alford  
2543850       Ricardo Allen  
2555162         Deion Jones  
2555255  De'Vondre Campbell  
...                     ...  
2553798         Andrew East  
2556814       Max McCaffrey  
2552576        Sean Mannion  
2561469    Darious Williams  
2558095       Malachi Dupre  

[1303 rows x 6 columns]

In [31]:
def changeWeight(oldWeight):
    return round(oldWeight/2.205)

In [33]:
players["weight"] = players["weight"].apply(changeWeight)

In [34]:
players

height  weight   birthDate             collegeName position  \
nflId                                                                  
2539334    1.83      86  1990-09-10              Washington       CB   
2539653    1.78      84  1988-11-01  Southeastern Louisiana       CB   
2543850    1.75      84  1991-12-18                  Purdue       SS   
2555162    1.85     103  1994-11-04         Louisiana State      MLB   
2555255    1.91     105  1993-07-01               Minnesota      OLB   
...         ...     ...         ...                     ...      ...   
2553798    1.88     107  09/17/1991              Vanderbilt       LS   
2556814    1.88      91  05/17/1994                    Duke       WR   
2552576    1.98     105  1992-04-25            Oregon State       QB   
2561469    1.75      85  1993-03-15      Alabama-Birmingham       CB   
2558095    1.88      89  10/12/1995                     LSU       WR   

                displayName  
nflId                        
2539334     Desmond Trufant  
2539653       Robert Alford  
2543850       Ricardo Allen  
2555162         Deion Jones  
2555255  De'Vondre Campbell  
...                     ...  
2553798         Andrew East  
2556814       Max McCaffrey  
2552576        Sean Mannion  
2561469    Darious Williams  
2558095       Malachi Dupre  

[1303 rows x 6 columns]

In [35]:
players.to_csv("../processed_data/clean/players.csv")